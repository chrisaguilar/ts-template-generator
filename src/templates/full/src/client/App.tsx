import * as React from 'react';
import * as ReactDOM from 'react-dom';

const styles = require('./static/global');

import { Main } from './Main/Main';

ReactDOM.render(
  <Main message='Hello, World!' />,
  document.getElementById('app')
);

// Accepts HMR updates from server
if ((module as any).hot)
  (module as any).hot.accept();
