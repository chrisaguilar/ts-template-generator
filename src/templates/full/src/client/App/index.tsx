import * as React from 'react';
const styles = require('./styles.scss');

import Main from '../Main';

export default class App extends React.Component<any, any> {
  render() {
    return (
      <div className={ styles.main }>
        <div className={ styles.text }>
          <h1>Hello, World!</h1>
          <Main message='how are you?' />
        </div>
      </div>
    );
  }
}
