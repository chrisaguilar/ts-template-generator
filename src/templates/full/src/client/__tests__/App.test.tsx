import { shallow } from 'enzyme';
import * as React from 'react';
import { create as snapshot } from 'react-test-renderer';

import App from '../App';

describe('<App />', () => {
  it('should render', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.length).toBeGreaterThan(0);
  });
  it('renders correctly', () => {
    const tree = snapshot(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
