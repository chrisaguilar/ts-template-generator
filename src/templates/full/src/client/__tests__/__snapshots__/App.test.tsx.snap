// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="main"
>
  <div
    className="text"
  >
    <h1>
      Hello, World!
    </h1>
    <div>
      <h3>
        how are you?
      </h3>
    </div>
  </div>
</div>
`;
